compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1035,12 +1039,70 @@ public void visit(BLangUserDefinedType userDefinedTypeNode) {
 
     @Override
     public void visit(BLangFunctionTypeNode functionTypeNode) {
-        List<BType> paramTypes = new ArrayList<>();
-        functionTypeNode.getParams().forEach(t -> paramTypes.add(resolveTypeNode((BLangType) t.getTypeNode(), env)));
-        BType retParamType = resolveTypeNode(functionTypeNode.returnTypeNode, this.env);
-        resultType = new BInvokableType(paramTypes, retParamType, null);
+        resultType = createInvokableType(functionTypeNode.getParams(), functionTypeNode.restParam,
+                functionTypeNode.returnTypeNode, Flags.asMask(functionTypeNode.flagSet), env);
     } 
 
+    public BInvokableType createInvokableType(List<? extends BLangVariable> paramVars, BLangVariable restVariable,
+                               BLangType retTypeVar, int flags, SymbolEnv env) {
+        List<BType> paramTypes = new ArrayList<>();
+        List<BVarSymbol> params = new ArrayList<>();
+
+        boolean foundDefaultableParam = false;
+        for (BLangVariable paramNode :  paramVars) {
+            BLangSimpleVariable param = (BLangSimpleVariable) paramNode;
+            BType type = resolveTypeNode(param.getTypeNode(), env);
+            paramNode.type = type;
+            paramTypes.add(type);
+
+            if (param.expr != null) {
+                foundDefaultableParam = true;
+            }
+
+            BVarSymbol symbol = new BVarSymbol(type.flags, names.fromIdNode(param.name),
+                    env.enclPkg.symbol.pkgID, type, env.scope.owner);
+            param.symbol = symbol;
+
+            if (param.expr == null && foundDefaultableParam) {
+                dlog.error(param.pos, REQUIRED_PARAM_DEFINED_AFTER_DEFAULTABLE_PARAM);
```

Comment:
we usually do DiagnosticCode.REQUIRED_PARAM_DEFINED_AFTER_DEFAULTABLE_PARAM

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/19976/commits/7432217a84fd35c7ae7682d1e9173e9416f9ccdc